### PR TITLE
Update iota.rs rev and participation error handling

### LINKED ISSUE #1803: Participation overview is empty even though participated
## Bug description

I have a wallet in which i'm participating in a proposal. If i look at the explorer, i see that i  indeed participate (https://explorer.shimmer.network/testnet/output/0x8ce2354a3fb8fa32ed18e4cb573bcac618d67f25e636564e20ff26c9135bef950000). But if i call `getParticipationOverview` for that wallet, i get `{participations: {}}`

it is reproducable if i recover the profile from the mnemonic

## Steps To reproduce the bug

Explain how the maintainer can reproduce the bug.

1. Recover profile with `embark suspect tenant spider this august flee apart pulse mammal volcano wall pitch neutral fun program paddle life attend knife amount crystal clay fringe`
2. select wallet with index 4 (has a balance of 1889.0654 SMR)
3. call `getParticipationOverview`
4. get `{participations: {}}`

## Expected behaviour

If i call `getParticipationOverview` for that wallet, i should get `{participations: {0x00d77d8fe83ed844ee67027c8d6e22da6a6ddbcb84dccedd0a1517d56e8ae258: { } }}`

## Actual behaviour

I get `{participations: {}}`

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Update iota.rs rev and participation error handling

## Links to any relevant issues

Fixes https://github.com/iotaledger/wallet.rs/issues/1803

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Running the voting example

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -87,6 +87,11 @@ pub enum Error {
     #[cfg(feature = "participation")]
     #[error("voting error {0}")]
     Voting(String),
+    /// Participation error
+    #[cfg(feature = "participation")]
+    #[error("participation error {0}")]
+    #[serde(serialize_with = "display_string")]
+    Participation(#[from] iota_client::api_types::plugins::participation::error::Error),
```
Comment (thibault-martinez) [inline src/error.rs:94]: Changelog

File: src/account/operations/transaction/high_level/burning_melting/burn_nft.rs [hunk 2]
```diff
@@ -35,7 +35,7 @@ impl AccountHandle {
                 &[Address::Nft(NftAddress::new(nft_id))],
                 &output_data,
                 current_time,
-                true,
+                Some(AliasTransition::State),
```
Comment (thibault-martinez) [inline src/account/operations/transaction/high_level/burning_melting/burn_nft.rs]: Why not None ?

File: src/account/operations/transaction/high_level/burning_melting/destroy_alias.rs [hunk 3]
```diff
@@ -41,7 +42,7 @@ impl AccountHandle {
                 &[Address::Alias(AliasAddress::new(alias_id))],
                 &output_data,
                 current_time,
-                true,
+                Some(AliasTransition::State),
```
Comment (thibault-martinez) [inline src/account/operations/transaction/high_level/burning_melting/destroy_alias.rs]: State ?

File: src/account/operations/transaction/high_level/burning_melting/burn_nft.rs (hunk 2 above)
Comment (Thoralf-M) [reply src/account/operations/transaction/high_level/burning_melting/burn_nft.rs:38]: Because we need a state transition to unlock the alias address

Comment (Thoralf-M) [reply src/account/operations/transaction/high_level/burning_melting/burn_nft.rs]: Thanks, thought we need the alias address to unlock

File: src/account/types/mod.rs [hunk 4]
```diff
@@ -62,11 +62,11 @@ impl OutputData {
         account: &Account,
         current_time: u32,
         bech32_hrp: &str,
-        alias_state_transition: bool,
+        alias_transition: Option<AliasTransition>,
```
Comment (thibault-martinez) [inline src/account/types/mod.rs:65]: Changelog because public

**Review by thibault-martinez — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**